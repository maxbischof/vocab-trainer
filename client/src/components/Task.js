import React, { useState } from 'react'
import styled from 'styled-components'

export default function Task({ word, onSolution }) {
  const [backgroundColor, setBackgroundColor] = useState()
  const [userInput, setUserInput] = useState('')

  function onSubmit(event) {
    event.preventDefault()
    const input = userInput.toLowerCase().trim()
    const reference = word.fields.foreignword.toLowerCase().trim()

    if (input === reference) {
      setBackgroundColor('var(--success)')
    } else {
      setBackgroundColor('var(--error)')
    }
    setTimeout(() => {
      setBackgroundColor('var(--background)')
    }, 1000)
    onSolution()
    setUserInput('')
  }

  return (
    <TaskContainer backgroundColor={backgroundColor}>
      <h2>{word.fields.germanword}</h2>
      <form onSubmit={onSubmit}>
        <input
          type="text"
          name="userTranslation"
          placeholder="Übersetze auf Französisch"
          onBlur={onSubmit}
          onChange={(e) => setUserInput(e.target.value)}
          value={userInput}
          autocomplete="off"
          autocorrect="off"
          autocapitalize="off"
          spellcheck="false"
        ></input>
      </form>
    </TaskContainer>
  )
}

const TaskContainer = styled.div`
  background: ${(props) => props.backgroundColor};
  height: 100vh;
  width: 100vw;
  padding 50px;
  ${(props) =>
    props.backgroundColor === 'var(--background)' &&
    'transition: background 1s ease-out;'}
`
