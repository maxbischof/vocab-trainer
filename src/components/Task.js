import React, { useState } from 'react'
import styled from 'styled-components'
import { evaluate } from '../lib/evaluation'

export default function Task({ word, onSolution, addAnswer }) {
  const [backgroundColor, setBackgroundColor] = useState()
  const [userInput, setUserInput] = useState('')

  function onSubmit(event) {
    event.preventDefault()
    addAnswer(userInput)

    if (evaluate(userInput, word.foreign) > 0) {
      setBackgroundColor('var(--success)')
    } else {
      setBackgroundColor('var(--error)')
    }
    setTimeout(() => {
      setBackgroundColor('var(--background)')
    }, 1000)
    onSolution()
    setUserInput('')
  }

  return (
    <TaskContainer backgroundColor={backgroundColor}>
      <h2>{word.native}</h2>
      <form onSubmit={onSubmit}>
        <input
          type="text"
          name="userTranslation"
          placeholder="Übersetze auf Französisch"
          onBlur={onSubmit}
          onChange={(e) => setUserInput(e.target.value)}
          value={userInput}
          autocomplete="off"
          autocorrect="off"
          autocapitalize="off"
          spellcheck="false"
        ></input>
      </form>
    </TaskContainer>
  )
}

const TaskContainer = styled.div`
  background: ${(props) => props.backgroundColor};
  height: 100vh;
  width: 100vw;
  padding: 50px;
  ${(props) =>
    props.backgroundColor === 'var(--background)' &&
    'transition: background 1s ease-out;'}
`
