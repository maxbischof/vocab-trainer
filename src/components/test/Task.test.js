import React from 'react'
import Task from './Task'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const word = { foreign: 'le ordinateur', native: 'der Computer' }
  const tree = renderer.create(<Task word={word} />).toJSON()
  expect(tree).toMatchSnapshot()
})
