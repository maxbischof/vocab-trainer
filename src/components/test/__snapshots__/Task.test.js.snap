// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ bIUVHg"
>
  <h2>
    der Computer
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      autocapitalize="off"
      autocomplete="off"
      autocorrect="off"
      name="userTranslation"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Übersetze auf Französisch"
      spellcheck="false"
      type="text"
      value=""
    />
  </form>
</div>
`;
